app/code/Magento/Store/App/Action/Plugin/Context.php

```diff
@@ -94,11 +96,10 @@ public function beforeDispatch(
         }
         try {
             $currentStore = $this->storeManager->getStore($storeCode);
+            $this->updateContext($currentStore);
         } catch (NoSuchEntityException $exception) {
             $this->processInvalidStoreRequested($exception);
         }
-
-        $this->updateContext($currentStore);
```

Comment:
What is the problem here?
If the `NoSuchEntityException` is thrown, then the `$this->updateContext($currentStore);` is never called.
Because `processInvalidStoreRequested` method throws another exception.

Could you please cover the corresponding case with an automated test?

Reply:
Hi @dmytro-ch : 
Thank you for your reply. Yes,it may never run if the block go to `NoSuchEntityException` . But why it below the `catch (NoSuchEntityException $exception) ` while it can only run if it run in "Try" block? It should move to the "Try" block. 

Thanks.

Reply:
I don't think that we need to put any other logic to `try` block in this particular case.
The main idea is that exceptions should not be recoverable in a way that it's done here. The logic must be linear as much as possible. Thus the `try` block should contain only the calls that throw an expected exception.

Also, the `$this->processInvalidStoreRequested();` can be called any time even before this try-catch block.
So this change doesn't actually affect anything.